sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3315,6 +3329,54 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
     }
   }
 
+  /**
+   * Resolves columns of an output table from the data in a logical plan. This rule will:
+   *
+   * - Insert casts when data types do not match
+   * - Detect plans that are not compatible with the output table and throw AnalysisException
+   */
+  object ResolveSetVariable extends Rule[LogicalPlan] {
+    override def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsWithPruning(
+      _.containsPattern(COMMAND), ruleId) {
+      case setVariable: SetVariable
+        if setVariable.sourceQuery.resolved && !setVariable.targetVariables.forall(_.resolved) =>
+
+        /**
+         * Resolve the left hand side of the SET
+         */
+        val resolvedVars = setVariable.targetVariables.map { variable =>
+            variable match {
```

Comment:
please fix ident (-2 spaces for each line)

Reply:
Done